### PR TITLE
gossipd: change the pending_cannouncement list to htable

### PR DESCRIPTION
**Pull request by trueptolemy**
fix `/* FIXME: Make this a htable! */` !

Change the pending_cannouncement (in `struct node`) from `struct list_head pending_cannouncement;` to `struct pending_cannouncement_map *pending_cannouncements;`

### CODE REVIEW

File: gossipd/routing.h [hunk 1]
```diff
@@ -198,9 +199,8 @@ struct routing_state {
 	/* node_announcements which are waiting on pending_cannouncement */
 	struct pending_node_map *pending_node_map;
 
-	/* FIXME: Make this a htable! */
 	/* channel_announcement which are pending short_channel_id lookup */
-	struct list_head pending_cannouncement;
+	struct pending_cannouncement_map *pending_cannouncements;
```
Comment (rustyrussell) [inline gossipd/routing.h]: Nice!  Only one trivial change: we don't need a pointer here, saving an allocation.

Comment (trueptolemy) [reply gossipd/routing.h]: Thank you! I've done it :)

BTW, when shouldn't I use the pointer for a htable?
There `nodes` and `pending_node_map` are both the pointer to htable. Should I also save them as allocations?
```
struct routing_state {
...
	struct node_map *nodes;

	struct pending_node_map *pending_node_map;

	struct pending_cannouncement_map pending_cannouncements;
...
};
```

Comment (rustyrussell) [reply gossipd/routing.h]: Yes!  Unless you try to convert it and realize there's some subtle reason I'm wrong :)

File: gossipd/routing.c [hunk 2]
```diff
@@ -832,19 +810,23 @@ static struct pending_cannouncement *
...
+	struct pending_cannouncement *pann;
 
-	list_for_each(&rstate->pending_cannouncement, i, list) {
-		if (short_channel_id_eq(scid, &i->short_channel_id))
-			return i;
+	for (pann = pending_cannouncement_map_first(&rstate->pending_cannouncements, &i);
+	     pann;
+	     pann = pending_cannouncement_map_next(&rstate->pending_cannouncements, &i)) {
+		if (short_channel_id_eq(scid, &pann->short_channel_id))
+			return pann;
```
Comment (rustyrussell) [inline gossipd/routing.c]: Oops, this iterates through the hash table!  You want:

```
pann = pending_cannouncement_map_get(&rstate->pending_cannouncements, scid);
```

Comment (trueptolemy) [reply gossipd/routing.c]: I'm so Sorry...I corrected it!

**Pull request merged**